### PR TITLE
UserDatabase Migration Tests: Part 7

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

Part 7 of #2699

Implemented integration tests for Notifications related tables for migration from version 126 to 127.

https://wearezeta.atlassian.net/browse/AN-6745
#### APK
[Download build #1622](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1622/artifact/build/artifact/wire-dev-PR2716-1622.apk)
[Download build #1623](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1623/artifact/build/artifact/wire-dev-PR2716-1623.apk)
[Download build #1629](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1629/artifact/build/artifact/wire-dev-PR2716-1629.apk)

### CODE REVIEW

File: storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/CloudNotificationsStatsTableTestHelper.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    companion object {
+
+        private const val CLOUD_NOTIFICATION_STATS_TABLE_NAME = "FCMNotificationStats"
+        private const val CLOUD_NOTIFICATION_STATS_STAGE_COL = "stage"
+        private const val CLOUD_NOTIFICATION_STATS_FIRST_BUCKET_COL = "bucket1"
+        private const val CLOUD_NOTIFICATION_STATS_SECOND_BUCKET_COL = "bucket2"
+        private const val CLOUD_NOTIFICATION_STATS_THIRD_BUCKET_COL = "bucket3"
+
+        fun insertCloudNotificationStat(stage: String, firstBucket: Int,
```
Comment (BradleyWire) [inline storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/CloudNotificationsStatsTableTestHelper.kt]: These can live on their own line each, it's more readable (and doesn't create too many lines as there are only five params). I can understand with larger functions.

File: storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/NotificationsTables126to127MigrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    private fun validateMigration() =
+        testHelper.validateMigration(
+            TEST_DB_NAME,
+            127,
+            true,
+            USER_DATABASE_MIGRATION_126_TO_127
+        )
+
+    private fun getUserDb() =
```
Comment (mejdoo) [inline storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/NotificationsTables126to127MigrationTest.kt]: @gizemb I will address your comments in part 6 once done with all the tables. We need to make sure asap that the migration of the tables is tested.

File: storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/NotificationsTables126to127MigrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        getUserDatabase(
+            getApplicationContext(),
+            TEST_DB_NAME,
+            UserDatabase.migrations
+        )
+
+    private suspend fun allCloudNotifications() =
+        getUserDb().cloudNotificationsDao().allCloudNotifications()
+
```
Comment (BradleyWire) [inline storage/src/androidTest/kotlin/com/waz/zclient/storage/userdatabase/notifications/NotificationsTables126to127MigrationTest.kt]: redundant whitespace for consistency

File: storage/src/main/kotlin/com/waz/zclient/storage/db/notifications/NotificationsDataEntity.kt [hunk 4]
```diff
@@ -5,7 +5,7 @@ import androidx.room.Entity
 import androidx.room.PrimaryKey
 
 @Entity(tableName = "NotificationData")
-data class NotificationsEntity(
+data class NotificationsDataEntity(
```
Comment (gizemb) [inline storage/src/main/kotlin/com/waz/zclient/storage/db/notifications/NotificationsDataEntity.kt]: Assuming that data contains info. of a single notification, NotificationData (singular) could be better

Comment (mejdoo) [reply storage/src/main/kotlin/com/waz/zclient/storage/db/notifications/NotificationsDataEntity.kt]: done

**Pull request merged**